torch/_inductor/dependencies.py

```diff
@@ -359,5 +359,50 @@ def extract_read_writes(
     )
 
 
+def extract_input_node_reduction_ranges(  # noqa: F722
+    input_node: ".ir.TensorBox",  # type: ignore[valid-type] # noqa: F722
+) -> Tuple[Optional[List[sympy.Expr]], Optional[List[sympy.Expr]]]:
+    """
+    Returns the size and reduction size of all inputs, if the sizes and reduction_sizes (if exist) are all the same.
+    It's possible that a node has multiple inputs, some are Reduction nodes and others are Pointwise nodes.
+    In this case, reduction_sizes of the Reduction nodes need to be the same.
+    Otherwise returns (None, None).
+    """
+
+    from .ir import ComputedBuffer, Loops
+
+    if not isinstance(input_node.data.data, Loops):
```

Comment:
I think we need some checks to ensure `.data` and `.data.data` exist.  There are some cases like views that result in different nesting.